src/react/account/AccountRoleSelectButtonAndModal.tsx

```diff
@@ -14,17 +19,38 @@ import { getRoleArnStored, setRoleArnStored } from '../utils/localStorage';
 import { Icon } from '../ui-elements/Help';
 import { AccountSelectorButton } from '../ui-elements/Table';
 
-function AccountRoleSelectButtonAndModal() {
+export function AccountRoleSelectButtonAndModal({
+  bigButton,
+  onClick,
+}: {
+  onClick?: () => void;
+  bigButton?: boolean;
+}) {
   const accounts = useAccounts();
   const { path } = useRouteMatch();
+  const location = useLocation();
+  const history = useHistory();
   const { account, selectAccountAndRoleRedirectTo } = useCurrentAccount();
   const { roleArn } = useDataServiceRole();
   const [assumedRoleArn, setAssumedRoleArn] = useState(roleArn);
   const [isModalOpen, setIsModalOpen] = useState(false);
   const accountName = account?.Name;
   const [assumedAccount, setAssumedAccount] = useState(accountName);
+  const redirectDataConsummers = useRedirectDataConsumers();
+
+  useEffect(() => {
+    if (location.search === '?assumeRole') {
+      setIsModalOpen(true);
+    }
+  }, [location.search]);
```

Comment:
Hm no, if that was the case it would've been working already with no intervention needed :) The problem is that when AccountRoleSelectButtonAndModal is rendered it's inside ReauthDialog. The moment in which ReauthDialog is closed, it is unmounted, thus unmounting AccountRoleSelectButtonAndModal as well. So the reason I'm doing this roundabout approach is to trigger an opening of the modal

Reply:
There's another caveat to the current approach though - if we're at a page where there's no `<AccountRoleSelectButtonAndModal />`.While working on the issue and testing various situations - it didn't happen, but it's possible that I missed some screen in which `<ReauthDialog />` shows up and the page under the modal doesn't have `<AccountRoleSelectButtonAndModal />` . A solution to this would be to redirect to a certain page - for example `/accounts?assumeRole` but I'm not sure which page would be a good candidate for that